### ISSUE TITLE
Python and Kernel versions should be kept in sync

### ISSUE THREAD

**Issue by cipri-tom** [microsoft/vscode-jupyter#6224]
Tried searching, couldn't find anything

## Expected behaviour

The Python version and the Kernel selector can be different. I believe this influences the completions offered.

![Screenshot 2021-06-09 at 11 27 57](https://user-images.githubusercontent.com/2991890/121330023-fd885a80-c915-11eb-80f2-7ddae94486d7.png)

Instead, I expect to have only one selector, and the version of python to be consistent, based on the kernel that I select.

## Logs
Will attach later if needed

## Environment data

-   VS Code version: 1.57.0-insider Commit: c7119ae8b6390a5700f6b28f0f1511d676c96562
-   Jupyter Extension version (available under the Extensions sidebar): v2021.6.914258744
-   Python Extension version (available under the Extensions sidebar): v2021.6.908196464-dev | Pylance v2021.6.1-pre.1
-   OS (Windows | Mac | Linux distro) and version: Mac 10.15.7
-   Python and/or Anaconda version: see below
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Remote, and window connected to remote host via Remote-SSH

**Comment by rchiodo** [comment]
Thanks for the suggestion. The kernel python version being different than your python interpreter is actually on purpose. This allows you to run multiple notebooks with different versions of python at the same time. 

There is a bug where the intellisense engine isn't using the kernel's python:
https://github.com/microsoft/vscode-jupyter/issues/4718

We could make an option though that forces the notebook to use the same python as the global python interpreter. I'll switch this to an enhancement to add that.

**Comment by joyceerhl** [comment]
@rchiodo At one point I think we used the active Python interpreter as the kernel for notebooks--did we do away with that?

**Comment by rchiodo** [comment]
> @rchiodo At one point I think we used the active Python interpreter as the kernel for notebooks--did we do away with that?

That  should be the case for a **new** notebook. At least when first created.

I believe @cipri-tom is asking for the python interpreter to always be in sync with any notebook. That there isn't two places to pick the python in use.

**Comment by joyceerhl** [comment]
Yeah, I only asked because in the screenshot above it looks like @cipri-tom has a new blank notebook open. But perhaps the kernel used did initially match the selected interpreter, and then they used the kernel selector to pick a different kernel?

**Comment by joyceerhl** [comment]
Regardless, this looks like a bug. We're supposed to hide the Python interpreter selector when the kernel status is displayed.

**Comment by cipri-tom** [comment]
@joyceerhl Today I'm working locally and I notice there is no Python interpreter selector. So this could be related to using the Remote server or the remote SSH extension (or both).

**Comment by DonJayamanne** [comment]
Team notes: I think we should close this issue as we no longer display the interpreter when the notebook is visible, or we should close this in favor of the other issues we have for getting started or kernel enumeration.
Hence moving to triage.

**Comment by DonJayamanne** [comment]
Closing for now based on previous comment and this issue https://github.com/microsoft/vscode-jupyter/issues/7171
